client/status_events.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2022 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0
+//
+// SPDX-License-Identifier: EPL-2.0
+
+package client
+
+import (
+	"sync"
+)
+
+// statusEventsConsumer uses a bounded cyclic events queue, safe for concurrent use.
+// When capacity is reached, oldest events in the queue are replaced by newer ones.
+type statusEventsConsumer struct {
```

Comment:
It's used as status event consumer but by definition it's more generic and better name should be used.